iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c

```diff
@@ -50,7 +50,11 @@ static IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_RESULT getDataCallback(IOTHUB_CLIENT_F
 
             if (block_count < 100)
             {
-                snprintf(data_to_upload, sizeof(data_to_upload), data_to_upload_format, block_count);
+                if (snprintf(data_to_upload, sizeof(data_to_upload), data_to_upload_format, block_count) < 0)
```

Comment:
should there also be a check that the written size < sizeof(data_to_upload)?

Reply:
?? 
snprintf will return less than zero if the written size (including null) is greater than sizeof(data_to_upload)
size is computed from strlen()(
??

Reply:
From here: https://www.cplusplus.com/reference/cstdio/snprintf/ (emphasis mine)

> The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is **non-negative and less than n**, the string has been completely written.

If it's more than n then it wasn't completely written?

Reply:
fixed